semeio/communication/reporter.py

```diff
@@ -0,0 +1,37 @@
+import json
+import os
+
+
+class FileReporter(object):
+    def __init__(self, output_dir):
+        self._output_dir = output_dir
+
+        if not os.path.isabs(output_dir):
+            err_fmt = "Expected output_dir to be an absolute path, received '{}'"
+            raise ValueError(err_fmt.format(output_dir))
+
+    def publish(self, namespace, data):
+        output_file = self._prepare_output_file(namespace) + ".json"
+
+        with open(output_file, "w") as f:
```

Comment:
For me, it is unclear as to why we are overwriting the data in a given namespace?

Reply:
Well, mostly due to the lack of a better alternative. Any other alternative is prone to give highly unexpected results. Note that as we are basing the location on `enspath`, users typically use `CASE_NAME` as part of this and hence would have separate locations for different cases. Also, note that the persistence of data is highly experimental and will not reside of disk for long and hence any attempt to pick up the data in the other end is at your own risk 😬